cmd/src/batch_preview.go

```diff
@@ -40,9 +41,20 @@ Examples:
 		}
 
 		ctx, cancel := contextCancelOnInterrupt(context.Background())
-		defer cancel()
 
-		if err = executeBatchSpec(ctx, executeBatchSpecOpts{
+		defer cancel(nil)
+
+		failFastCancel := func(error) {}
+		if flags.failFast {
+			failFastCancel = cancel
+		}
+
+		cctx := executor.CancelableContext{
+			Context: ctx,
+			Cancel:  failFastCancel,
+		}
```

Comment:
Is there a need for this struct? Is there a reason we can't pass in the context and `failFastCancel` as separate arguments?

Or better still have the `failFastCancel` as one of the fields in `executeBatchSpecOpts`, that way we are certain `ctx` is always going to be of type `context.Context` and wouldn't be confusing to anyone in the future;